torch/_dynamo/eval_frame.py

```diff
@@ -1229,17 +1229,34 @@ def patch():
                 DistributedDataParallel._inside_ddp_forward, recursive=False
             )
 
-        from ..optim import adagrad, adam, adamax, adamw, asgd, nadam, sgd
+        from ..optim import (
```

Comment:
Should we just import all the optims here? There's also sparse_adam, radam, lbfgs, rmsprop. Since there's already an hasattr check, this would be more reliable + correct.

e.g., why is radam excluded? it has foreach as well

Reply:
lol yeah I'll import all of them, not sure where I got this list from haha.

Reply:
Step is disabled at 1243 for these since they are unsupported. I think I need to add rmsprop though.

Reply:
Is there harm in adding them anyway? What if one day we add support?